presto-druid/src/main/java/com/facebook/presto/druid/DruidConnectorFactory.java

```diff
@@ -48,7 +53,13 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
         try {
             Bootstrap app = new Bootstrap(
                     new JsonModule(),
-                    new DruidModule(context.getTypeManager()));
+                    new DruidModule(), binder -> {
+                binder.bind(TypeManager.class).toInstance(context.getTypeManager());
+                binder.bind(FunctionMetadataManager.class).toInstance(context.getFunctionMetadataManager());
+                binder.bind(RowExpressionService.class).toInstance(context.getRowExpressionService());
+                binder.bind(LogicalRowExpressions.class).toInstance(new LogicalRowExpressions(context.getRowExpressionService().getDeterminismEvaluator(), context.getStandardFunctionResolution(), context.getFunctionMetadataManager()));
```

Comment:
Kinda weird. `LogicalRowExpressions` can be created anytime than an injected singleton

Reply:
Comment not addressed?